pkg/services/ngalert/schedule/compat.go

```diff
@@ -84,6 +91,31 @@ func noDataAlert(labels data.Labels, annotations data.Labels, alertState *state.
 	}
 }
 
+// errorAlert is a special alert sent when evaluation of an alert rule failed due to an error. Like noDataAlert, it
+// replaces the old behaviour of "Keep Last State" creating a separate alert called DatasourceError.
+func errorAlert(labels, annotations data.Labels, alertState *state.State, urlStr string) *models.PostableAlert {
+	if name, ok := labels[model.AlertNameLabel]; ok {
+		labels[Rulename] = name
+	}
+	labels[model.AlertNameLabel] = ErrorAlertName
+
+	var queryError expr.QueryError
+	if errors.As(alertState.Error, &queryError) {
+		labels["refID"] = queryError.RefID
+		annotations["error"] = queryError.Error()
+	}
```

Comment:
it looks like you add duplicated labels and annotations here. state manager adds the same labels. 
- probably it does not make sense to populate labels here. 
- If there is a reason to keep them in both places, then I would recommend creating the label names in some shared place and using it in both places.

Reply:
You are right, I've moved the annotations and labels to the state package!